DataProcessing/SmartInsiderConverter.cs

```diff
@@ -265,6 +395,37 @@ private void WriteToFile<T>(DirectoryInfo destinationDirectory, Dictionary<strin
             }
         }
 
+        /// <summary>
+        /// Writes to a temp file and moves the content to the final universe directory
+        /// </summary>
+        /// <param name="destinationDirectory">Directory to write final file to</param>
+        /// <param name="contents">Contents to write to file</param>
+        private void WriteUniverseFile(DirectoryInfo destinationDirectory, Dictionary<string, Dictionary<string, string>> contents)
+        {
+            Parallel.ForEach(contents, kvp =>
+            {
+                var date = kvp.Key;
+
+                var finalPath = Path.Combine(destinationDirectory.FullName, "universe", $"{kvp.Key}.csv");
+                var finalFileExists = File.Exists(finalPath);
+
+                var lines = new HashSet<string>(kvp.Value.Select(kv => $"{kv.Key},{kv.Value}"));
+                if (finalFileExists)
+                {
+                    foreach (var line in File.ReadAllLines(finalPath))
+                    {
+                        lines.Add(line);
```

Comment:
If there is a difference in value between the new line and the existing line for the same date and ticker, which will it use? Should probably use the new line 👍 
- Suggest removing the temp file handling bellow and just directly writing to the destination
- `var date = kvp.Key;` is not used